src/System Application/App/Cryptography Management/src/RSAImpl.Codeunit.al

```diff
@@ -0,0 +1,256 @@
+    var
+        Verified: Boolean;
+    begin
+        if Bytes.Length() = 0 then
+            exit(false);
+        Verified := TryVerifyData(Bytes, HashAlgorithm, RSASignaturePadding, Signature);
+        if not Verified and (GetLastErrorText() <> '') then
+            Error(GetLastErrorText());
+        exit(Verified);
+    end;
+
+    [TryFunction]
+    [NonDebuggable]
+    local procedure TryVerifyData(Bytes: DotNet Array; HashAlgorithm: Enum "Hash Algorithm"; RSASignaturePadding: Enum "RSA Signature Padding"; Signature: DotNet Array)
+    var
+        DotNetHashAlgorithmName: DotNet HashAlgorithmName;
+        DotNetRSASignaturePadding: DotNet RSASignaturePadding;
+    begin
+        HashAlgorithmEnumToDotNet(HashAlgorithm, DotNetHashAlgorithmName);
+        RSASignaturePaddingToDotNet(RSASignaturePadding, DotNetRSASignaturePadding);
+        if not DotNetRSA.VerifyData(Bytes, Signature, DotNetHashAlgorithmName, DotNetRSASignaturePadding) then
+            Error('');
+    end;
+    #endregion
+
+    #region Encryption & Decryption
+    [NonDebuggable]
+    procedure Encrypt(XmlString: SecretText; PlainTextInStream: InStream; OaepPadding: Boolean; EncryptedTextOutStream: OutStream)
+    var
+        PlainTextBytes: DotNet Array;
+        EncryptedTextBytes: DotNet Array;
+        DotNetRSAEncryptionPadding: DotNet RSAEncryptionPadding;
+    begin
+        FromSecretXmlString(XmlString);
+        InStreamToArray(PlainTextInStream, PlainTextBytes);
+
+        if OaepPadding then
+            DotNetRSAEncryptionPadding := DotNetRSAEncryptionPadding.OaepSHA256
+        else
+            DotNetRSAEncryptionPadding := DotNetRSAEncryptionPadding.Pkcs1;
+
+        EncryptedTextBytes := DotNetRSA.Encrypt(PlainTextBytes, DotNetRSAEncryptionPadding);
+        ArrayToOutStream(EncryptedTextBytes, EncryptedTextOutStream);
+    end;
+
+    [NonDebuggable]
+    procedure Decrypt(XmlString: SecretText; EncryptedTextInStream: InStream; OaepPadding: Boolean; DecryptedTextOutStream: OutStream)
+    var
+        EncryptedTextBytes: DotNet Array;
+        DecryptedTextBytes: DotNet Array;
+        DotNetRSAEncryptionPadding: DotNet RSAEncryptionPadding;
+    begin
+        FromSecretXmlString(XmlString);
+        InStreamToArray(EncryptedTextInStream, EncryptedTextBytes);
+        if OaepPadding then
+            DotNetRSAEncryptionPadding := DotNetRSAEncryptionPadding.OaepSHA256
+        else
+            DotNetRSAEncryptionPadding := DotNetRSAEncryptionPadding.Pkcs1;
+        DecryptedTextBytes := DotNetRSA.Decrypt(EncryptedTextBytes, DotNetRSAEncryptionPadding);
+        ArrayToOutStream(DecryptedTextBytes, DecryptedTextOutStream);
+    end;
+    #endregion
+
+
+    #region XmlString
+#if not CLEAN24
+    [NonDebuggable]
+    [Obsolete('Replaced by ToSecretXmlString with SecretText data type for XmlString.', '24.0')]
```

Comment:
Not sure if the build passes with tag `24.0`. If it doesn't, it needs to be bumped to `25.0`

Reply:
Same for the one below.
I'd like to outright not include them in the PR, but since this needs to implement, the only SignatureAlgorithm interface, it'll just have to stay even if not used..